### ISSUE TITLE
style.grid-template-columns inline binding doesn't work

### ISSUE THREAD

**Issue by maxisam** [angular/angular#28897]
# 🐞 bug report

### Description
style.grid-template-columns inline binding doesn't work

```
<div class="grid" [style.grid-template-columns]="'repeat(5,  1fr)'">
	<div>test</div>
	<div>test</div>
	<div>test</div>
	<div>test</div>
	<div>test</div>
</div>
```

## 🔬 Minimal Reproduction

https://stackblitz.com/edit/angular-hbqtjz

## 🌍  Your Environment

**Angular Version:**
<pre><code>7.1.4</code></pre>

**Anything else relevant?**

It works by using ngStyle like  `[ngStyle]="{ 'grid-template-columns': 'repeat(5,  1fr)'} `

**Comment by ericmartinezr** [comment]
Dup of/related to https://github.com/angular/angular/issues/9343

**Comment by Splaktar** [comment]
This doesn't seem like a dupe, certainly related though. The other issue is specific to CSS Variables/Properties.

This issue is just using plain CSS ([grid-template-columns](https://developer.mozilla.org/en-US/docs/Web/CSS/grid-template-columns)) along with its related CSS Function [repeat](https://developer.mozilla.org/en-US/docs/Web/CSS/repeat).

**Comment by k-arraytom** [comment]
I'm trying to bind the number of rows to a component property. 
[ngStyle]="{ 'grid-template-columns': 'repeat(someProperty, 1fr)'} does not work.
Any suggestions?
Thanx in advanx

**Comment by matteobortolazzo** [comment]
Hi, any update on this? 
I would like to bind ` repeat(auto-fill, minmax(${this.itemMinWidth}, 1fr))` but it's not working.

**Comment by jonrimmer** [comment]
@matteobortolazzo In Angular v9 they have at least fixed binding custom properties, so although although the sanitizer is still broken w.r.t grid functions, you may be able to move the repeat definition into your component's stylesheet, and supply just the variable using a style binding, as follows:

```scss
.container {
  grid-template columns: repeat(auto-fill, minmax(var(--itemMinWidth, 1fr));
}
```

```html
<div class="container" [style.--itemMinWidth]="itemMinWidth">
   ...
</div>
```

**Comment by andreElrico** [comment]
this seems to be working for me:

```
@HostBinding('style.grid-template-columns') get hb_gridCols() { return `repeat(${this.gridCols}, 1fr)`; }
@HostBinding('style.grid-template-rows') get hb_gridRows() { return `repeat(${this.gridRows}, 1fr)`; }
@HostBinding('style.grid-gap.px') get hb_gridGapPx() { return '10'; }
```

**Comment by Vandivier** [comment]
Do we know why the binding doesn't work in some cases?

I see style bindings label, so I'm not sure, but this may be an alternative solution:
https://github.com/angular/angular/issues/36650

**Comment by weixinwu** [comment]
Is there any other alternative workaround?
I tried @jonrimmer solution, the repeat() seems not getting the variable.
Here is my code:
`  grid-template-columns: repeat(var(--numOfDays), 1fr);`
`<div class="grid-container" [style.--numOfDays]="itemWidth"></div>`

**Comment by Jakartoine** [comment]
🎉  Acceptable Workaround 🎉

I've got rid of this problem in Angular 8 by injecting DOCUMENT token and
set manually my style property. (should still be OK for Angular 9 & 10)

**Document injection:**

![image](https://user-images.githubusercontent.com/27675999/88784601-80864380-d190-11ea-9df7-d8c31d07c347.png)

**Style property assignment:**

![image](https://user-images.githubusercontent.com/27675999/88784764-b4f9ff80-d190-11ea-942e-7e54fc7af87a.png)

**Use your style property inside your attached css file:**

![image](https://user-images.githubusercontent.com/27675999/88784865-d0fda100-d190-11ea-9720-97bd5db49854.png)

**Comment by joaodezembro** [comment]
>  Acceptable Workaround 
> 
> I've got rid of this problem in Angular 8 by injecting DOCUMENT token and
> set manually my style property. (should still be OK for Angular 9 & 10)
> 
> **Document injection:**
> 
> ![image](https://user-images.githubusercontent.com/27675999/88784601-80864380-d190-11ea-9df7-d8c31d07c347.png)
> 
> **Style property assignment:**
> 
> ![image](https://user-images.githubusercontent.com/27675999/88784764-b4f9ff80-d190-11ea-942e-7e54fc7af87a.png)
> 
> **Use your style property inside your attached css file:**
> 
> ![image](https://user-images.githubusercontent.com/27675999/88784865-d0fda100-d190-11ea-9720-97bd5db49854.png)

Works very well!

Thanks

**Comment by intellix** [comment]
Interestingly I'm having a problem with using the shorthand gridTemplate in ngStyle:

Works:
```html
<div class="d-grid" style='grid-template: "a . b" ". . ." 1rem "c c c"'>...</div>
```

Doesn't work: 
```ts
const ngStyle: Partial<CSSStyleDeclaration> = {
  gridTemplate: '"a . b" ". . ." 1rem "c c c"'
}
```

```html
<div class="d-grid" [ngStyle]="ngStyle">...</div>
```

It'll actually convert it to `grid-template-areas` which isn't the shorthand version and my 1rem gets clipped off: 

```html
<div class="d-grid" style="grid-template-areas: "a . b" ". . ." "c c c"">...</div>
```

**Comment by SimSla** [comment]
For anyone else who couldn't get the variable binding (https://github.com/angular/angular/issues/28897#issuecomment-557105277) to work, part of the problem appears to be that Angular converts variables declared inline to snake-case.

So `[style.--camelCase]="foo"` creates a css variable called `--camel-case`, which likely doesn't match what your stylesheet was expecting.

Using snake-case solved the problem for me without having to do anything like document injection.

**Comment by crisbeto** [comment]
Is this still a problem? The original example seems to work for me in Angular 10: https://stackblitz.com/edit/angular-ivy-xoluea?file=src%2Fapp%2Fapp.component.html

**Comment by wupaz** [comment]
For me grid-row-template & grid-column-template work for me but grid.row and grid.column didn't

**Comment by intellix** [comment]
From my above comment, it works now, but it still fails on Universal/Domino as it strips both singular/double quotes

**Comment by pkozlowski-opensource** [comment]
The original issue reported here is already fixed as starting from #35621 we no longer sanitize styles: https://stackblitz.com/edit/angular-ivy-bshted?file=src%2Fapp%2Fapp.component.ts

I'm not clear about what is going on with the Universal/Domino: if anyone is still facing problems in this area please open a new issue with a reproduce scenaril.
